### DISCUSSION TITLE
UPDATE queries on Table functions

### DISCUSSION THREAD

**Discussion by prashanthellina** [duckdb/duckdb#14695] [Q&A]
I have written a table function using the C-API that returns data. I would like to be able to get a callback to a C function when an `UPDATE` query is executed against the table function. Is this possible to accomplish with existing functionality?

e.g.
```sql
UPDATE my_table_function() SET col1 = 10 WHERE col2 > 1;
```

Thank you!

**Comment by prashanthellina** [comment]
From @Maxxen ([discord](https://discord.com/channels/909674491309850675/1148659944669851849/1304409022903877743))

Table functions arent tables, so you cant update or insert into them.
Maybe you want to look into creating a catalog extension to expose custom tables? Similar to how the postgres and mysql extensions work. Although thta is not supported in the C-api (and probably wont be for quite a while)

**Comment by Tishj** [comment]
I don't know what the effect of the example would be?
Does it change data in the `my_table_function()` so the next time it's called the returned data is different?
Does it just alter the returned data from this one call, essentially just creation a projection then?

**Reply by prashanthellina** [reply]
The former. i.e. Change the state so that next time when it is called the data reflects the previous update.

The use-case is that I embedded DuckDB in my C application where I use table functions to expose some in-memory datastructures so I can perform analysis using SQL queries. I would like to be able to change the in-memory datastructure state to alter application behavior too.
